mux_broker_test.go randomly failed due to race condition

mux_broker_test.go failed and print following logs. This fail happened randomly, after I restart the ci job, it passed.

```
==================
WARNING: DATA RACE
Write at 0x00c000119a51 by goroutine 90:
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:830 +0x174
Previous write at 0x00c000119a51 by goroutine 31:
  testing.(*common).FailNow()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:589 +0x4c
  testing.(*common).Fatalf()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:634 +0x94
  github.com/hashicorp/packer/packer/rpc.TestMuxBroker.func1()
      /home/travis/gopath/src/github.com/hashicorp/packer/packer/rpc/mux_broker_test.go:27 +0x1b2
Goroutine 90 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:878 +0x650
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:827 +0x162
  testing.runTests()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:1034 +0x2ee
  main.main()
      _testmain.go:98 +0x221
Goroutine 31 (finished) created at:
  github.com/hashicorp/packer/packer/rpc.TestMuxBroker()
      /home/travis/gopath/src/github.com/hashicorp/packer/packer/rpc/mux_broker_test.go:20 +0x2d6
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:827 +0x162
==================
--- FAIL: TestMuxBroker (0.00s)
    mux_broker_test.go:27: err: session shutdown
    testing.go:771: race detected during execution of test
```

_Originally posted by @chhaj5236 in https://github.com/hashicorp/packer/pull/7050#issuecomment-444372827_

--- a/packer/rpc/mux_broker_test.go
+++ b/packer/rpc/mux_broker_test.go
@@ -18,7 +18,7 @@ func TestMuxBroker(t *testing.T) {
 	go bc.Run()
 	go bs.Run()
 
-	errChan := make(chan error, 1)
+	errChan := make(chan error, 2)
 	go func() {
 		defer close(errChan)
 		c, err := bc.Dial(5)

--- a/packer/rpc/mux_broker_test.go
+++ b/packer/rpc/mux_broker_test.go
@@ -20,18 +20,16 @@ func TestMuxBroker(t *testing.T) {
 
 	errChan := make(chan error, 1)
 	go func() {
+		defer close(errChan)
 		c, err := bc.Dial(5)
 		if err != nil {
 			errChan <- fmt.Errorf("err dialing: %s", err.Error())
-			close(errChan)
 			return
 		}
 
 		if _, err := c.Write([]byte{42}); err != nil {
 			errChan <- fmt.Errorf("err writing: %s", err.Error())
 		}
-
-		close(errChan)
 	}()
 
 	client, err := bs.Accept(5)

--- a/packer/rpc/mux_broker_test.go
+++ b/packer/rpc/mux_broker_test.go
@@ -1,6 +1,7 @@
 package rpc
 
 import (
+	"fmt"
 	"net"
 	"testing"
 
@@ -21,13 +22,13 @@ func TestMuxBroker(t *testing.T) {
 	go func() {
 		c, err := bc.Dial(5)
 		if err != nil {
-			errChan <- fmt.Errorf("err dialing: %s", err)
+			errChan <- fmt.Errorf("err dialing: %s", err.Error())
 			close(errChan)
 			return
 		}
 
 		if _, err := c.Write([]byte{42}); err != nil {
-			errChan <- fmt.Errorf("err writing: %s", err)
+			errChan <- fmt.Errorf("err writing: %s", err.Error())
 		}
 
 		close(errChan)
@@ -51,8 +52,9 @@ func TestMuxBroker(t *testing.T) {
 		err, open := <-errChan
 		if !open {
 			if err != nil {
-				t.Fatalf(err)
+				t.Fatalf(err.Error())
 			}
+			break
 		}
 	}
 }

--- a/packer/rpc/mux_broker_test.go
+++ b/packer/rpc/mux_broker_test.go
@@ -17,15 +17,20 @@ func TestMuxBroker(t *testing.T) {
 	go bc.Run()
 	go bs.Run()
 
+	errChan := make(chan error, 1)
 	go func() {
 		c, err := bc.Dial(5)
 		if err != nil {
-			t.Fatalf("err: %s", err)
+			errChan <- fmt.Errorf("err dialing: %s", err)
+			close(errChan)
+			return
 		}
 
 		if _, err := c.Write([]byte{42}); err != nil {
-			t.Fatalf("err: %s", err)
+			errChan <- fmt.Errorf("err writing: %s", err)
 		}
+
+		close(errChan)
 	}()
 
 	client, err := bs.Accept(5)
@@ -40,6 +45,15 @@ func TestMuxBroker(t *testing.T) {
 
 	if data[0] != 42 {
 		t.Fatalf("bad: %d", data[0])
+	}
+
+	for {
+		err, open := <-errChan
+		if !open {
+			if err != nil {
+				t.Fatalf(err)
+			}
+		}
 	}
 }
 

--- a/packer/rpc/mux_broker_test.go
+++ b/packer/rpc/mux_broker_test.go
@@ -46,13 +46,9 @@ func TestMuxBroker(t *testing.T) {
 		t.Fatalf("bad: %d", data[0])
 	}
 
-	for {
-		err, open := <-errChan
-		if !open {
-			if err != nil {
-				t.Fatalf(err.Error())
-			}
-			break
+	for err := range errChan {
+		if err != nil {
+			t.Fatalf(err.Error())
 		}
 	}
 }

